Use mermaid for exercise flowchart.

mo-fu opened the pull request:
Add mermaid description for the exercise overview flowchart, removed the old `.png` and its coresponding `.dot` file.

osma commented:
It looks great, but it's quite large and perhaps a bit unwieldy on the page:

![kuva](https://user-images.githubusercontent.com/1132830/186362076-f133926e-52c7-456d-8c1b-058ee65fa538.png)

Is there any chance to move it to the left side to align better with the rest of the content?

Is it possible to reduce the vertical space between the nodes by, say, one third?

mo-fu commented:
I changed the spacing between nodes. Also changed the arrow shapes to `linear` as the angle between line and arrow head looked weird, Unfortunately I could not find a way to align the chart.

osma commented:
Looks much better now! It's more compact so doesn't break the layout too much. Great work!
Too bad we can't left-align it but OK. I think it's best to merge this now. We will still need to tweak it later as we add more exercises.
